### ISSUE TITLE
Docs request: does `pre-commit` with custom `additional_dependencies` need to re-specify `uv`

### ISSUE THREAD

**Issue by jamesbraza** [henryiii/check-sdist#58]
Firstly, this is a cool and super useful tool, just discovered it today 👍 

I saw in the "quick start" that it instructs to use `uv`: https://github.com/henryiii/check-sdist/tree/v1.0.0?tab=readme-ov-file#quick-start

And I set up my `pre-commit` hook like so (note I use `setuptools`):

```yaml
---
repos:
  - repo: https://github.com/henryiii/check-sdist
    rev: v1.0.0
    hooks:
      - id: check-sdist
        args: [--inject-junk]
        additional_dependencies: [setuptools>=64, setuptools_scm>=8]
```

I observed the `pre-commit` hook opts into `uv`: https://github.com/henryiii/check-sdist/blob/main/.pre-commit-hooks.yaml#L10

Do you mind adding a sentence to the `README.md` if I need to re-specify `uv` in my `additional_dependencies` because I have my own `additional_dependencies`?

In other words, should it be:

```yaml
        additional_dependencies: [setuptools>=64, setuptools_scm>=8]
        # or
        additional_dependencies: [setuptools>=64, setuptools_scm>=8, uv]
```

**Comment by henryiii** [comment]
uv is not needed if you are not using the `isolated` hook. The normal hook uses pre-commit to prepare the build environment so it gets cached, so uv isn't used (unless you install [pre-commit-uv](https://pypi.org/project/pre-commit-uv/, which hacks pre-commit to use uv for everything).

The isolated one does opt-in to uv since it's doing the environment setup every time. But users don't need to add deps if they use that.

`pypa/build` is used to do the build, though in theory now that uv supports building directly, that could be used instead. 

A bit more clarity in the docs would be good!
